CONTRIBUTING.md

```diff
@@ -4,50 +4,99 @@ Thank you for wanting to help out!
 Cuprate is in the stage where things are likely to change quickly, so it's recommended
 you ask questions in our public [Matrix room](https://matrix.to/#/#cuprate:monero.social).
 
-- [1. Submitting a pull request](#1-submitting-a-pull-request)
-	- [1.1 Rust toolchain](#11-rust-toolchain)
-	- [1.2 Draft PR](#12-draft-pr)
-	- [1.3 Passing CI](#13-passing-ci)
-	- [1.4 Ready for review](#14-ready-for-review)
-- [2. Crate names](#2-crate-names)
-- [3. Coding guidelines](#3-coding-guidelines)
-- [4. Keeping track of issues and PRs](#4-keeping-track-of-issues-and-prs)
+- [1. Submitting an issue](#1-submitting-an-issue)
+	- [1.1 Discussion](#11-discussion)
+	- [1.2 Proposal](#12-proposal)
+	- [1.3 Tracking issue](#13-tracking-issue)
+- [2. Submitting a pull request](#2-submitting-a-pull-request)
+	- [2.1 Rust toolchain](#21-rust-toolchain)
+	- [2.2 Draft PR](#22-draft-pr)
+	- [2.3 Passing CI](#23-passing-ci)
+	- [2.4 Ready for review](#24-ready-for-review)
+- [3. Keeping track of issues and PRs](#3-keeping-track-of-issues-and-prs)
+	- [3.1 Labels](#31-labels)
+	- [3.2 Tracking issues](#32-tracking-issues)
+- [4. Coding guidelines](#4-coding-guidelines)
+	- [4.1 General guidelines](#41-general-guidelines)
+	- [4.2 Crate names](#42-crate-names)
 - [5. Documentation](#5-documentation)
 - [6. Books](#6-books)
 	- [6.1 Architecture book](#61-architecture-book)
 	- [6.2 Protocol book](#62-protocol-book)
 	- [6.3 User book](#63-user-book)
 
-## 1. Submitting a pull request
-Once you have found something you would like to work on by:
+## 1. Submitting an issue
+Before starting work, consider opening an issue for discussion.
+
+If you have a plan already, you can jump straight into [submitting a pull request](#2-submitting-a-pull-request).
+
+Otherwise, see below for issue types and what they're used for.
+
+### 1.1 Discussion
+These are for general discussion on topics that have questions that aren't fully answered yet.
+
+If you would like to discuss a topic and get some feedback, consider [opening a discussion](https://github.com/Cuprate/cuprate/issues/new/choose).
+
+Examples:
+- https://github.com/Cuprate/cuprate/issues/40
+- https://github.com/Cuprate/cuprate/issues/53
+- https://github.com/Cuprate/cuprate/issues/163
+
+### 1.2 Proposal
+These are formal issues that specify changes that are _almost_ ready for implementation.
+
+These should answer some basic questions:
+- **What** is this proposal for?
+- **Why** is this proposal needed?
+- **Where** will this proposal make changes to?
+- **How** will this proposal be implemented?
+
+If you have a close to fully fleshed out idea, consider [opening a proposal](https://github.com/Cuprate/cuprate/issues/new/choose).
+
+Opening a PR with the `A-proposal` tag and writing the proposal in the PR description is also viable.
```

Comment:
Actually, can regular accounts even add labels to their own issues/PRs?

Reply:
Removing this for now.

Reply:
I don't think so